test/components/util/util.go

```diff
@@ -186,3 +192,108 @@ func filterNonControllerPods(pods *corev1.PodList) *corev1.PodList {
 	pods.Items = filteredPods
 	return pods
 }
+
+type PortForwardInfo struct {
```

Comment:
It seems this struct could be split into 2, so one represents user configuration and the other one is generated with `PortForward()`. This way, the API is clear, which fields should be set by the user and which fields will be set by the creating method. That could be taken even further, and this new struct could be private and only expose the `interface` outside of the package.

Reply:
I think creating interface is far fetched because we will be creating API in the blind without knowing who all will be consuming it and what parts need abstracting. I would avoid doing any such premature abstraction and wait until we have such a use case.

---

About creating two structs, what is wrong with having the distinction on the basis of private and public fields of the struct? Isn't it serving the same purpose?

Reply:
You're right. I didn't notice, that other fields are not exported. All good then.